src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -475,59 +452,64 @@ WEAVE_ERROR LoggingManagement::SetExchangeManager(nl::Weave::WeaveExchangeManage
  * @brief
  *   LoggingManagement constructor
  *
- * For prioritization to work correctly, inBuffers must be incrementally
- * increasing in priority.
+ * Initialize the LoggingManagement with an array of LogStorageResources.  The
+ * array must provide a resource for each valid importance level, the elements
+ * of the array must be in increasing numerical value of importance (and in
+ * decreasing importance); the first element in the array corresponds to the
+ * resources allocated for the most critical events, and the last element
+ * corresponds to the least important events.
  *
  * @param[in] inMgr         WeaveExchangeManager to be used with this logging subsystem
  *
- * @param[in] inNumBuffers  Number of buffers to use for event storage
- *
- * @param[in] inBufferLengths Description of inBufferLengths
- *
- * @param[in] inBuffers     The buffers to use for actual event logging.
+ * @param[in] inNumBuffers  Number of elements in inLogStorageResources array
  *
- * @param[in] inCounterKeys Keys naming persisted counters
+ * @param[in] inLogStorageResources  An array of LogStorageResources for each importance level.
  *
- * @param[in] inCounterEpochs An array of epochs for each of the persisted counters
- *
- * @param[in] inCounterStorage Application-provided storage for the persistent counters
- *
- * @return LoggingManagement
  */
-LoggingManagement::LoggingManagement(WeaveExchangeManager * inMgr, size_t inNumBuffers, size_t * inBufferLengths, void ** inBuffers,
-                                     nl::Weave::Platform::PersistedStorage::Key * inCounterKeys, const uint32_t * inCounterEpochs,
-                                     nl::Weave::PersistedCounter ** inCounterStorage)
+
+LoggingManagement::LoggingManagement(nl::Weave::WeaveExchangeManager * inMgr,
+                                     size_t inNumBuffers,
+                                     const LogStorageResources * const inLogStorageResources)
 {
     CircularEventBuffer * current = NULL;
     CircularEventBuffer * prev    = NULL;
     CircularEventBuffer * next    = NULL;
-    size_t i;
+    size_t i, j;
+
+    VerifyOrDie(inNumBuffers > 0);
 
     mThrottled   = 0;
     mExchangeMgr = inMgr;
 
-    for (i = 0; i < inNumBuffers; i++)
+    for (j = 0; j < inNumBuffers; j++)
     {
-        next = ((i + 1) < inNumBuffers) ? static_cast<CircularEventBuffer *>(inBuffers[i + 1]) : NULL;
+        i = inNumBuffers - 1 - j;
```

Comment:
If the order of elements is not important then "i' can be used directly and "j' is not needed.

Reply:
Order remains important, and the function expects the elements to be ordered.  By providing the importance, we admit to gaps and non-contiguous importance enums. documentation remains correct.